### PR TITLE
Rahul/ifl 2044 1 add rust method to split a sapling key into shares for FROST

### PR DESCRIPTION
**Pull request by patnir**
## Summary

Adding a method to split a sapling key. 
Defining the ShareSecretConfig

These two items are dependencies for splitting the spender key which will be next

## Testing Plan
Tests pass
## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-rust/src/transaction/mod.rs [hunk 1]
```diff
@@ -59,9 +71,6 @@ mod utils;
 mod value_balances;
 mod version;
```
Comment (jowparks) [inline ironfish-rust/src/transaction/mod.rs:73]: What was this line removed for?

File: ironfish-rust/src/transaction/mod.rs [hunk 2]
```diff
@@ -935,3 +947,44 @@ pub fn batch_verify_transactions<'a>(
...
+        config.min_signers,
+        identifiers,
+        rng,
+    )?;
+
+    for (_k, v) in shares.clone() {
+        frost::keys::KeyPackage::try_from(v)?;
+    }
+
+    let mut key_packages: HashMap<_, _> = HashMap::new();
```
Comment (hughy) [inline ironfish-rust/src/transaction/mod.rs:982]: should this `HashMap` use the specific `Identifier, KeyPackage` type? does it make a difference?

**Review by jowparks — APPROVED** [review]

**Review by hughy — COMMENTED** [review]
this looks good!

wondering if this should go into `keys/mod.rs` instead of `transaction/mod.rs`?

File: ironfish-rust/src/transaction/mod.rs (hunk 2 above)
Comment (jowparks) [reply ironfish-rust/src/transaction/mod.rs:982]: Yea you can remove the `: HashMap<_, _>` and it will do type inference

**Pull request merged**